src/yz_exchange_fsm.erl

```diff
@@ -122,27 +143,41 @@ prepare_exchange(start_exchange, S) ->
 prepare_exchange(timeout, S) ->
     do_timeout(S).
 
-update_trees(start_exchange, S=#state{yz_tree=YZTree,
-                                      kv_tree=KVTree,
+update_trees(start_exchange, S=#state{kv_tree=KVTree,
+                                      yz_tree = YZTree,
                                       index=Index,
                                       index_n=IndexN}) ->
-
-    update_request(yz_index_hashtree, YZTree, Index, IndexN),
-    update_request(riak_kv_index_hashtree, KVTree, Index, IndexN),
+    Self = self(),
+    update_request(
+        riak_kv_index_hashtree, KVTree, Index, IndexN,
+        fun() ->
+            yz_solrq_drain_mgr:drain([
+                {?EXCHANGE_FSM_PID, Self},
+                {?YZ_INDEX_HASHTREE_PARAMS, {YZTree, Index, IndexN}},
+                {?DRAIN_PARTITION, Index}
+            ])
+        end
+    ),
     {next_state, update_trees, S};
 
+update_trees({drain_error, Reason}, S) ->
+    lager:debug("Drain failed with reason ~p", [Reason]),
```

Comment:
This probably needs to be a lager:error call, not just a debug. We want to log when drains failed, especially because they will have caused the yz_solrq_drain_fsm to be killed if the timeout/cancel cycle takes too long. May want to take into account different error types (not log in_progress?) but need to make sure we can understand what's going on when something ends up in the crash log. /cc @kesslerm